src/test/java/com/squareup/javapoet/TypeSpecTest.java

```diff
@@ -54,6 +55,13 @@ private TypeElement getElement(Class<?> clazz) {
     return compilation.getElements().getTypeElement(clazz.getCanonicalName());
   }
 
+  private boolean isJava8() {
+    String version = System.getProperty("java.version");
+    int pos = version.indexOf('.');
+    pos = version.indexOf('.', pos+1);
+    return Double.valueOf(version.substring (0, pos)).compareTo(Double.valueOf(1.8)) >= 0;
```

Comment:
I think you can probably just do string comparison. The right thing will happen:
```
  return System.getProperty("java.version").compareTo("1.8") >= 0;
```